Issue/1732 empty product variations

anitaa1990 opened the pull request:
Fixes #1732. 

This PR fixes 2 issues:
- Variations section in Product Detail page is only displayed if `numVariations` is > 0.
- If all variations of a product is deleted from the web while in the Variation list screen in app and PTR is initiated, the an empty view is displayed.

#### Screenshots
<img width="300" src="https://user-images.githubusercontent.com/22608780/71335969-43f3c900-256b-11ea-95ec-e5afc4820562.png">

#### Testing
##### Issue 1
- Delete existing variations of a product.
- Open a variable product from the app and ensure that the `variations` section is **not** displayed.

##### Issue 2
- Add variations to a product.
- Open a variable product from the app and ensure that the `variations` section is displayed.
- Click on the `Variations` section and ensure that variation list is displayed.
- Delete the variations of the product from `wp-admin`.
- Pull to refresh in the app once the variations are deleted.
- Ensure that empty view is displayed in the app.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

nbradbury commented:
@anitaa1990 Looks good except for one edge case: if you view variations in the app then delete them all on the web and PTR, the variation list correctly shows the empty view. But when you return to product detail, the variations still appear. Also, there are merge conflicts that need to be addressed.

anitaa1990 commented:
Forgot to push these changes last night.

> if you view variations in the app then delete them all on the web and PTR, the variation list correctly shows the empty view. But when you return to product detail, the variations still appear.

Nice catch! This was because there was no option to remove a view once it was added. Fixed in 524fc4e

> Also, there are merge conflicts that need to be addressed.

Addressed in ef045e6

Also found a bug where the variations list was displaying the empty view if loadMore returned empty data. Fixed in 3c28f88

Ready for another round 👍
